src/components/listing-search-page/Card/index.jsx

```diff
@@ -28,39 +37,111 @@ import {
  * @return {JSX.Element}
  */
 function Card({ data }) {
+  const { features, agent } = data;
+  const listingBy = `Listing By : ${agent.roles[0]?.role} / ${
+    agent.roles[1]?.role ? agent.roles[1]?.role : ''
+  }`;
+  const navigate = useNavigate();
   // eslint-disable-next-line react/prop-types
   if (data.length < 0) return null;
 
-  const { images, price, address, fullBathrooms, bedrooms } = data;
-
+  const handleClick = () => {
+    navigate(`${listingPaths.details}/${data.id}`);
+  };
   return (
-    <CardContainer>
+    <CardContainer onClick={handleClick}>
       <ImageContainer>
         <Badge>Sale</Badge>
-        <Image src={images[0]} />
+        <Image src={data.images[0]} />
       </ImageContainer>
 
       <CardText>
-        <PriceText>$ {price}</PriceText>
+        <PriceText>$ {data.price}</PriceText>
         <LocationText>
           <LocationIcon>
             <PinIcon />
           </LocationIcon>
-          <LocationIcon>{address}</LocationIcon>
+          <LocationIcon>{data.address}</LocationIcon>
         </LocationText>
-        <Label> Listing By:</Label>
+        <Label>{listingBy}</Label>
+
         <IconsContainer>
-          <IconsNumber>
-            {bedrooms}
-            <Bed />
-          </IconsNumber>
-          <WifiIcon />
-          <IconsNumber>
-            {fullBathrooms}
-            <BathtubIcon />
-          </IconsNumber>
-          <AirConditionerIcon />
-          <BenchIcon />
+          {data.bedrooms ? (
+            <IconsNumber>
+              {data.bedrooms}
+              <Bed />
+            </IconsNumber>
+          ) : null}
```

Comment:
Same for all other elements please 
```suggestion
          {bedrooms && (
            <IconsNumber>
              {data.bedrooms}
              <Bed />
            </IconsNumber>
          ) }
```